connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java

```diff
@@ -329,11 +331,25 @@ static ConfigDef getConfigDefFromTransformation(String key, Class<?> transformat
         if (transformationCls == null || !Transformation.class.isAssignableFrom(transformationCls)) {
             throw new ConfigException(key, String.valueOf(transformationCls), "Not a Transformation");
         }
+        if (Modifier.isAbstract(transformationCls.getModifiers())) {
+            String[] childClasses = Stream.of(transformationCls.getClasses())
+                .filter(transformationCls::isAssignableFrom)
+                .filter(c -> !Modifier.isAbstract(c.getModifiers()))
+                .filter(c -> Modifier.isPublic(c.getModifiers()))
+                .map(Class::getName)
+                .toArray(String[]::new);
```

Comment:
What about just making `childClasses` a `Set<String>`, since `String.join(...)` takes an `Iterable`? Arguably it's not much different, but it's more in line with other uses of `String.join`.

Or, maybe better yet, what about just joining with the stream? Something like:
```
            String childClassNames = Stream.of(transformationCls.getClasses())
                                           .filter(transformationCls::isAssignableFrom)
                                           .filter(c -> !Modifier.isAbstract(c.getModifiers()))
                                           .filter(c -> Modifier.isPublic(c.getModifiers()))
                                           .map(Class::getName)
                                           .collect(Collectors.joining(", "));
            String message = childClassNames.trim().isEmpty() ?
                             "Transformation is abstract and cannot be created." :
                             "Transformation is abstract and cannot be created. Did you mean " + childClassNames + "?";
```

Reply:
That works for me. I'll change it.